WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -187,14 +187,40 @@ extension PointOfSaleAggregateModel {
 extension PointOfSaleAggregateModel {
     func barcodeScanned(_ barcode: String) {
         Task {
-            let placeholderItemID = cart.addLoadingItem()
+            let placeholderItemID = cart.addLoadingItem().id
+
+            analytics.track(
+                event: .PointOfSale.addItemToCart(
+                    sourceViewType: .scanner,
+                    itemType: .loading
+                )
+            )
```

Comment:
I have second thoughts about tracking this.

It will cause barcode scanning stores to show twice as many items being added to the cart, which could skew other measurements in future. This is a UI artefact, and each one will either be replaced with a product or an error...

I can imagine it helping a little with debugging, in the case that it _isn't_ replaced... but it doesn't seem like it would be helpful enough to justify itself.

We could also do some duration tracking... but that could be done without the loading state.

I'll raise it in P2 so we can sync with Android.

Reply:
Yes, valid concern. We could have a separate `scan` event, then either `item_added` or `scan_error`.